cast/src/lib.rs

```diff
@@ -447,8 +453,40 @@ impl SimpleCast {
         format!("0x{}", s)
     }
 
+    pub async fn generate_interface(
+        address_or_path: String,
+        chain: Chain,
+        etherscan_api_key: Option<String>,
+    ) -> Result<Vec<InterfaceSource>> {
+        let mut contract_names: Vec<String> = Vec::new();
+        let contract_abis: Vec<Abi> = if Path::new(&address_or_path).exists() {
+            println!("Reading ABI from local file at {}", &address_or_path);
+            contract_names.push("Interface".to_owned());
+            let file = std::fs::read_to_string(&address_or_path).expect("unable to read abi file");
+            vec![serde_json::from_str(&file).expect("unable to parse json ABI")]
+        } else {
+            println!("Requesting ABI from etherscan for contract at: {}", &address_or_path);
+            let client = Client::new(chain, etherscan_api_key.unwrap())?;
+            let contract_source =
+                client.contract_source_code(address_or_path.parse().unwrap()).await?;
```

Comment:
avoid unwraps or make it an expect(..)

```suggestion
                client.contract_source_code(address_or_path.parse()?).await?;
```